src/Pim/Bundle/CatalogBundle/Doctrine/ORM/CompletenessRemover.php

```diff
@@ -123,17 +135,48 @@ protected function bulkRemoveCompletenesses(CursorInterface $products)
                 $bulkCounter++;
             }
 
-            $product->getCompletenesses()->clear();
+            $this->clearProductCompleteness($product, $channel, $locale);
         }
 
         if (!empty($productIds)) {
-            $statement->bindValue('productIds', $productIds, Type::SIMPLE_ARRAY);
-            $statement->execute();
+            $this->entityManager->getConnection()->executeQuery(
+                $query,
+                [$productIds],
+                [Connection::PARAM_INT_ARRAY]
+            );
             $this->indexer->indexAll($bulkedProducts);
         }
     }
 
     /**
+     * @param ProductInterface      $product
+     * @param null|ChannelInterface $channel
+     * @param null|LocaleInterface  $locale
+     */
+    protected function clearProductCompleteness(
+        ProductInterface $product,
+        ChannelInterface $channel = null,
+        LocaleInterface $locale = null
+    ) {
+        if (null === $channel && null === $locale) {
+            $product->getCompletenesses()->clear();
+
+            return;
```

Comment:
If there is no locale nor channel, then we need to remove all completeness. I prefer perform a `clear` then an early return, rather than perform an `ArrayCollection::filter()` in all cases.